nncf/common/pruning/operations.py

```diff
@@ -204,6 +204,10 @@ def generate_output_mask(cls, node: NNCFNode, graph: NNCFGraph,
         if not not_empty_masks:
             return None
 
+        # if node is leaf node return None
+        if not graph.get_output_edges(node):
+            return None
+
```

Comment:
Looks like problem is more fundamental: some of the input node has non empty pruning masks. The result of such return could lead to inconsistent pruning layout. @sooahleex, let's try to inspect this graph together